kernel/Kconfig.vm

```diff
@@ -98,13 +98,19 @@ config KERNEL_DIRECT_MAP
 
 endif # KERNEL_VM_SUPPORT
 
+config CPU_USE_MMU
+	bool
+	help
+	  This option is enabled when the CPU's memory management unit is active
+	  and the arch_mem_map() API is available.
+
 menuconfig MMU
 	bool "MMU features"
-	depends on CPU_HAS_MMU
+	depends on CPU_USE_MMU
+	default y
```

Comment:
This default can be removed, but in that case `Kconfig.mem` would need to remain unchanged (approximately) and still imply `MMU` and `DEMAND_PAGING`.